### ISSUE TITLE
Feature Request: Add MultiGoshape for goshapes that need to share path, but with different options (work-around exist)

### ISSUE THREAD

**Issue by GGAlanSmithee** [daleblackwood/goshapes#27]
I propose to add a new node; `MultiGoshape`, which would be somewhat similar to a `MultiShaper`, but with the important difference that you can have different `PathOptions`, but share the same `GoPath`. This would allow for things such as;

* a layered road
* a river with banks in the side of it
* a forest that follows a road

There exist a work-around for this now, which is to save the `GoPath` and re-use it, but the UX for that is not perfect, because all references does not update when you save one of them (this is a common occurance in general though.

Here's an example of a forest that follows the same `GoPath` as the road. Settings like this makes sense if you restrict movement to the road, and the camera can never pan outside of a certain perimeter. So this was achieved by sharing the `GoPath`, using two different `Goshape`s.

![image](https://github.com/user-attachments/assets/13a2ff6c-34a6-4c32-a3e8-220b4ec44d46)

**Comment by daleblackwood** [comment]
Another great idea - it's difficult to decide how to implement it. Perhaps a per-shaper path-options override?

**Comment by GGAlanSmithee** [comment]
Yeah, it's not super clear how to implement this. I do believe the existing work-around makes this non-essential. It's more of a QOL thing. For now, saving and re-using the `path` works (with the before-mentioned quirk that you have to trigger an update for all affected shapers)
